pos_product_cost_security/models/ir_http.py

```diff
@@ -0,0 +1,16 @@
+# Copyright 2021 Tecnativa - David Vidal
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from odoo import models
+
+
+class Http(models.AbstractModel):
+    _inherit = "ir.http"
+
+    def session_info(self):
+        """This context comes later from UI to the rpc call"""
+        result = super(Http, self).session_info()
```

Comment:
Can we avoid this override in so such high level?

Reply:
I saw no other way to inject into the user_context to get it with the rpc call: https://github.com/odoo/odoo/blob/14.0/addons/point_of_sale/static/src/js/models.js#L655

Reply:
But you talk about having already an existing context when on PoS. I think we can re-use it without fear of not having side effects (just bypass that security, not doing `sudo`).

@Yajo do you know anyway other way to inject variables in session context?

Reply:
Ok, I got a cleaner workaround to avoid injecting contexts all around